src/facility.c

```diff
@@ -33,6 +34,15 @@ stumpless_get_facility_string( enum stumpless_facility facility ) {
   }
   return "NO_SUCH_FACILITY";
 }
+
+enum stumpless_facility
+stumpless_get_facility_enum( const char *facility_string ) {
+  for (int i = 0; i < sizeof facility_enum_to_string / sizeof facility_enum_to_string[0]; i++)
+    if (strcmp(facility_string, facility_enum_to_string[i]) == 0)
+      return i * sizeof facility_enum_to_string[0];
```

Comment:
This multiplication is incorrect, but I suspect it appears to work because the size of a pointer happens to be 8 on all of the current builds as they are 64-bit. I'll make a note to add in a CI build with different sizes to catch potential issues like this in the future.

In the meantime, please change this to `i << 3` , as is used in other facility value creations (see `include/stumpless/facility.h`).

Reply:
yes, it also seemed strange to me, but I replicated the behavior from the "stumpless_get_facility_string" function above, which is defined:
```
 const char *
stumpless_get_facility_string( enum stumpless_facility facility ) {
  size_t facility_upper_bound = sizeof facility_enum_to_string;
  if ( !facility_is_invalid(facility) ) {
    return facility_enum_to_string[facility/ sizeof facility_enum_to_string[0]];
  }
  return "NO_SUCH_FACILITY";
}
```
the return is doing the same operation in reverse. I'll change that function as well, since I'm pretty sure it has the same error, but please let me know if I'm missing something and I'll revert it.

Reply:
You're absolutely right, that function has the same issue as well! A different size architecture build is much needed, it seems. Thank you for finding/correcting those issues as well.

Reply:
no problem at all, thanks for the detailed feedback.